### PR TITLE
Fixed isMoveAllowed

### PR DESCRIPTION
**Pull request by codingmarisa**
The spaces of immovable objects can now no longer be occupied by the player.

### CODE REVIEW

File: include/room_objects/Room.hpp [hunk 1]
```diff
@@ -69,6 +70,7 @@ class Room
   string getDescription();
   Door *getDoor(int doorIndex);
   Door **getDoors();
+  int getItemsSize();
```
Comment (TuckerDane) [inline include/room_objects/Room.hpp]: getItemArraySize()

File: include/room_objects/Room.hpp [hunk 2]
```diff
@@ -38,6 +38,7 @@ class Room
   string type;
   string name;
   string description;
+  int itemsArrayIndex;
```
Comment (TuckerDane) [inline include/room_objects/Room.hpp]: itemsArraySize

File: src/game/Update.cpp [hunk 3]
```diff
@@ -58,16 +58,17 @@ bool Game::isMoveAllowed(int y, int x)
     else // if the space contains an item that you can walk on
     {
         Item **items = rooms[player.getCurrentRoom()]->getItems();
-        for (int i = 0; i < rooms[player.getCurrentRoom()]->getMaxItems(); i++)
+        if (items[0] != NULL)
         {
-            if (items[i] != NULL)
+            int itemsSize = rooms[player.getCurrentRoom()]->getItemsSize();
+            for (int i = 0; i < itemsSize; i++)
```
Comment (TuckerDane) [inline src/game/Update.cpp]: itemsArraySize

**Comment by codingmarisa** [comment]
Good catch! I'll update that!

**Pull request closed without merging**